src/common/exceptions/http-error.filter.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  HttpStatus,
+  Logger,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Response, Request } from 'express';
+import {
+  DevelopmentErrorResponse,
+  HttpExceptionResponse,
+} from '../interfaces/http-exception-response.interface';
+
+@Catch()
+export class HttpErrorFilter implements ExceptionFilter {
+  private logger: Logger;
+  constructor(private configService: ConfigService) {
+    this.logger = new Logger();
+  }
+  catch(exception: Error, host: ArgumentsHost): any {
+    const ctx = host.switchToHttp();
+    const request = ctx.getRequest<Request>();
+    const response = ctx.getResponse<Response>();
+
+    let statusCode: HttpStatus;
+    let errorMessage: string;
+
+    if (exception instanceof HttpException) {
+      statusCode = exception.getStatus();
+      const errorResponse = exception.getResponse();
+      errorMessage =
+        (errorResponse as HttpExceptionResponse).error || exception.message;
+    } else {
+      statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+      errorMessage = 'Internal server error occurred!!';
+    }
+
+    const developmentErrorResponse = this.getDevelopmentErrorResponse(
+      statusCode,
+      errorMessage,
+      request,
+    );
+
+    const errorLog = this.getErrorLog(
+      developmentErrorResponse,
+      request,
+      exception,
+    );
+
+    this.logger.error(errorLog);
+
+    const environmentMode = this.configService.get('NODE_ENV');
+    response
+      .status(statusCode)
+      .json(
+        environmentMode === 'dev' || environmentMode === 'development'
```

Comment:
```suggestion
        environmentMode !== 'production' 
```

mungkin better seperti ini kondisinya

Reply:
itu 2 kondisi karena sebelumnya di env saya sama kang @fajarhikmal214  gak sama penulisan valuenya, jadi dibikin 2 kondisi seperti itu, dan sekarang sudah dibikin validasi untuk value environmentnya .

oke thanks masukannya kang @ayocodingit